dev/release/verify-release-candidate.sh

```diff
@@ -499,6 +499,41 @@ maybe_setup_conda() {
   fi
 }
 
+install_maven() {
+  MAVEN_VERSION=3.8.7
+  if command -v mvn > /dev/null; then
+    SYSTEM_MAVEN_VERSION=$(mvn -v | head -n 1 | awk '{print $3}')
+    show_info "Found Maven version ${SYSTEM_MAVEN_VERSION} at $(command -v mvn)."
+  else
+    SYSTEM_MAVEN_VERSION=0.0.0
+    show_info "Maven installation not found."
+  fi
+
+  if [[ "$MAVEN_VERSION" == "$SYSTEM_MAVEN_VERSION" ]]; then
+    show_info "System Maven version ${SYSTEM_MAVEN_VERSION} matches required Maven version ${MAVEN_VERSION}. Skipping installation."
+  else
+    older_version=$(printf '%s\n%s\n' "$SYSTEM_MAVEN_VERSION" "$MAVEN_VERSION" | sed 's/$/|/' | sort -V | sed 's/|$//' | head -n1)
```

Comment:
```suggestion
    # Append pipe character to make preview release versions like "X.Y.Z-beta-1" sort as
    # older than their corresponding release version "X.Y.Z". This works because 
    # `sort -V` orders the pipe character lower than any character in version numbers.
    older_version=$(printf '%s\n%s\n' "$SYSTEM_MAVEN_VERSION" "$MAVEN_VERSION" | sed 's/$/|/' | sort -V | sed 's/|$//' | head -n1)
```

Reply:
Is this sufficient @kou?

Reply:
Yes. Thanks.

BTW, we will never specify unreleased version to `MAVEN_VERSION` (required minimal Maven version). So the situation will be never happen. For example, Maven 4.0.0 is an unreleased version now ( https://maven.apache.org/docs/history.html ). We should not use 4.0.0 for `MAVEN_VERSION` because it requires unreleased (not the "official") Maven for verifiers.

Reply:
Yes, but if someone is using this script to do release verification and they have a preview release of the required Maven version installed, then without this, the release verification script would not install the newer Maven release.

Maybe this will never happen but it seems possible.